DEVELOPER_GUIDE.md

```diff
@@ -0,0 +1,300 @@
+# Harvester Developer Guide
+
+This guide provides essential information for developers who want to contribute to Harvester.
+
+## Table of Contents
+
+- [Things You Need to Know Before Development](#things-you-need-to-know-before-development)
+  - [Understanding Harvester](#understanding-harvester)
+  - [Rules](#rules)
+- [Repository Relations](#repository-relations)
+  - [Main Components](#main-components)
+  - [Add-ons](#add-ons)
+  - [Rancher Related](#rancher-related)
+  - [Integrated Upstream Repositories](#most-closely-integrated-upstream-repositories)
```

Comment:
it seems that this anchor link `#most-closely-integrated-upstream-repositories` is invalid